src/main/java/org/testng/DataProviderInformation.java

```diff
@@ -0,0 +1,33 @@
+package org.testng;
+
+import java.lang.reflect.Method;
+
+/**
+ * Represents information pertaining to a data provider that would be invoked.
+ */
+public class DataProviderInformation {
+    private Object instance;
+    private Method method;
+
+    public DataProviderInformation(Method method, Object instance) {
```

Comment:
I noticed the reflection part but it is useless extra work for users because I think they will need information from the annotation.
And as we already have the same need, it is a perfect way to use it too.